scheduler/scheduler.go

```diff
@@ -227,30 +233,117 @@ func (driver *MesosSchedulerDriver) masterDetected(master *mesos.MasterInfo) {
 		}
 
 		driver.MasterPid = pid // save for downstream ops.
+		driver.tryAuthentication()
+	} else {
+		log.Infoln("No master detected.")
+	}
+}
+
+func (driver *MesosSchedulerDriver) tryAuthentication() {
+	if driver.authenticated {
+		// programming error
+		panic("already authenticated")
```

Comment:
I'm trying to use panics in places where I'd otherwise use assertions: I really do want the func to completely fail if the assertion fails.

What do you think about adding a panic handler in the messenger, so that panics in the scheduler driver implementation don't always crash the messenger delivery loop?